pkg/master/controller/crdregistration/construction.go

```diff
@@ -0,0 +1,324 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package crdregistration
```

Comment:
this factoring is not a good idea. The master should not know about CRDs. CRDs should be normal resources, in this case provided by a delegated API server.

I could imagine to have very minimal CRD support here in the master. But all the code related to openapi certainly belong into apiextensions-apiserver, not into master.

Reply:
While going through the current code: we already have this minimal crd controller. But we should keep pushing logic into apiextensions-apiserver as far as possible. This swagger code certainly belongs there and not into the kube-apiserver/master.

Reply:
I agree. Moved to `apiextensions-apiserver/pkg/openapi/`